### PR TITLE
lesson04: mailroom2, trigrams resubmission

### PR DESCRIPTION
**Pull request by bromleysj**

### CODE REVIEW

**Comment by bromleysj** [comment]
re: lesson05; I commented out in mailroom3.py the code for creating a wrapper function (safe_input).  I repeatedly received a NameError and would like some more direction on this.  
Also, please provide feedback on exceptions; I struggled somewhat with this topic.

File: students/scott_bromley/lesson04/mailroom2.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    :return: response to question/prompt
+    """
+    return input(prompt).strip()
+
+
+if __name__ == "__main__":
+    print("Running", __file__)
+    main()
+else:
+    print("Running %s as imported module", __file__)
```
Comment (charlesrobison) [inline students/scott_bromley/lesson04/mailroom2.py]: Nice updates - works well!  Grade adjusted

File: students/scott_bromley/lesson04/trigrams.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    except IndexError:
+        print("You must pass in a filename")
+        sys.exit(1)
+    in_data = read_in_data(filename)
+    words = make_words(in_data)
+    word_pairs = build_trigram(words)
+    new_text = build_text(word_pairs)
+    print(new_text)
+else:
+    print("Running %s as imported module", __file__)
```
Comment (charlesrobison) [inline students/scott_bromley/lesson04/trigrams.py]: This works well too - nice update.  Please submit this on the course site for grading.

File: students/scott_bromley/lesson05/mailroom3.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                                 statistics.mean(donors.get(donor))])
+        except IndexError as donor_summary_err:
+            raise donor_summary_err
+        else:
+            donor_summary.sort(key=lambda x: float(x[1]), reverse=True)
+            print(f"Donor Name{'':<20} | Total Given{'':>0} | Num Gifts{'':>0} | Average Gift{'':>0}")
+            print(f"-" * 72)
+        finally:
+            for name, total, num_gift, avg_gift in donor_summary:
+                print(f"{name:<32}${total:>11.2f}{num_gift:>12}  ${avg_gift:>13.2f}")
```
Comment (charlesrobison) [inline students/scott_bromley/lesson05/mailroom3.py]: Getting multiple outputs every time it cycles through the donor db

**Comment by bromleysj** [comment]
I've fixed the issue with multiple prints, however, I'd like some guidance
on how to implement a wrapper for input.

I did comment this out in my code, but I really want to learn how to use
this.  I know that I need the "@" notation on the function I want to wrap,
so get_user_input should get the @safe_input, but I continually get an
error when I run my code.

Thanks,

Scott

**Pull request merged**